core/src/main/java/org/elasticsearch/index/query/BaseQueryParserTemp.java

```diff
@@ -33,7 +33,7 @@
 
     @Override
     public QueryBuilder fromXContent(QueryParseContext parseContext) throws IOException, QueryParsingException {
-        Query query = parse(parseContext);
+        Query query = parse(parseContext.context());
```

Comment:
@javanna This is one of the places where the new parse context still needs to refer back to the enclosed context, since it is used in fromXContent here temporarily. The whole class will finally got however.